fog/ledger/enclave/impl/src/lib.rs

```diff
@@ -49,9 +79,17 @@ impl ReportableEnclave for SgxLedgerEnclave {
     }
 }
 
-impl LedgerEnclave for SgxLedgerEnclave {
+impl<OSC> LedgerEnclave for SgxLedgerEnclave<OSC>
+where
+    OSC: ORAMStorageCreator<StorageDataSize, StorageMetaSize>,
+{
     fn enclave_init(&self, self_id: &ResponderId) -> Result<()> {
         self.ake.init(Default::default(), self_id.clone())?;
+        let mut lk = self.keyimagestore.lock()?;
+        let desired_capacity: u64;
+        desired_capacity = 1024 * 1024; // process num of users samples just like the enclave view
```

Comment:
this is no bueno because this is hard-coding a limit of 1 million key images into the view enclave

the way the view enclave currently handles this is there is a view-enclave init params, and the untrusted side can determine this number without an enclave upgrade

i will send a link

Reply:
https://github.com/mobilecoinfoundation/fog/blob/792802edf580be3fb9000b41b306b297ae16e6f1/fog/view/enclave/impl/src/lib.rs#L80